src/kibana/components/doc_table/components/table_header.js

```diff
@@ -12,10 +16,19 @@ define(function (require) {
         columns: '=',
         sorting: '=',
         indexPattern: '=',
+        type: '='
       },
       template: headerHtml,
       controller: function ($scope) {
-
+        $scope.downloadQueueManager = DownloadQueueManager;
+        $scope.captureSelectModalManager = CaptureSelectModalManager;
+        
+        if ($scope.type && ($scope.type === 'savedsearch')){
+            // sorry... had to grab panel name from parent (x8) scope :(
+            $scope.tableID = $scope.$parent.$parent.$parent.$parent.$parent.$parent.$parent.$parent.panel.id;
```

Comment:
ouch.
OK. Can you write a comment that is in sync with the ```parent``` to show where all these ancestral history is leading us?